<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="LivingNorwayR">
<title>EML R package overview • LivingNorwayR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="EML R package overview">
<meta property="og:description" content="LivingNorwayR">
<meta property="og:image" content="https://livingnorway.github.io/LivingNorwayR/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">LivingNorwayR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/EML_R_packages_overview.html">EML R package overview</a>
    <a class="dropdown-item" href="../articles/LNWorkshopExample_TOV-E.html">Handling Darwin Core Files With Living Norway:  Example Using the TOV-E Dataset</a>
    <a class="dropdown-item" href="../articles/LNWorkshopExample_TOV-E_Metadata.html">LNWorkshopExample_TOV-E_Metadata</a>
    <a class="dropdown-item" href="../articles/Mapping_data_to_Darwin_Core_using_the_LivingNorwayR_package.html">Mapping data to Darwin Core using the LivingNorwayR package</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/LivingNorway/LivingNorwayR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="EML_R_packages_overview_files/htmlwidgets-1.6.2/htmlwidgets.js"></script><link href="EML_R_packages_overview_files/vis-9.1.0/vis-network.min.css" rel="stylesheet">
<script src="EML_R_packages_overview_files/vis-9.1.0/vis-network.min.js"></script><script src="EML_R_packages_overview_files/visNetwork-binding-2.1.2/visNetwork.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>EML R package overview</h1>
                        <h4 data-toc-skip class="author">Matt</h4>
            
            <h4 data-toc-skip class="date">04 12, 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/LivingNorway/LivingNorwayR/blob/HEAD/vignettes/EML_R_packages_overview.Rmd" class="external-link"><code>vignettes/EML_R_packages_overview.Rmd</code></a></small>
      <div class="d-none name"><code>EML_R_packages_overview.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="r-packages-that-interact-with-eml">R packages that interact with EML<a class="anchor" aria-label="anchor" href="#r-packages-that-interact-with-eml"></a>
</h2>
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p>EML (Ecological Metadata Language; <span class="citation">(M. Jones
et al. 2019)</span>) is a version of the machine and human readable XML
(Extensible Markup Language) which is designed to facilitate data
documentation (i.e. metadata) to ease open data and sharing. XML is a
language related to HTML, but was designed to tag document content and
allows for validation against specific schemas. XML is well supported in
many different programming languages and can be readily translated to
other formats <span class="citation">(McCartney and Jones
2002)</span>.</p>
<p>EML consists of “modules” (with separate schemas) that allow the
description of data attributes (e.g. spatial, temporal, taxonomic
extent). EML defines several resource types (e.g. “dataset”) that
inherit a common set of elements or “tags” which facilitate discovery of
metadata. One of the biggest advantages of EML is that it is extensible
with additional modules easily integrated <span class="citation">(McCartney and Jones 2002)</span>.</p>
<p>GBIF (the Global Biodiversity Information Facility), which is an
international network and research infrastructure, provides open access
to data about biodiversity. The GBIF metadata profile is based on EML
and as such anyone interacting with GBIF needs to be able to read, write
or visualise data in the EML format.</p>
<p>A serious barrier to data sharing is the learning curve and time cost
needed to produce metadata. EML is a complex set of elements (many 100’s
of them are available) most of which are not relevant to any single
dataset <span class="citation">(McCartney and Jones 2002)</span>.
Writing directly in EML can be a technical barrier to many researchers
who, in ecology, may only have experience in the R language which has a
very different syntax and data structure. The challenge is, therefore,
to make filling in metadata as easy as possible for all researchers
regardless of their data type.</p>
<p>Several R packages have functions to read, write or interact with EML
files and here we aim to identify those that will be useful for
researchers working with ecological data.</p>
<div class="section level4">
<h4 id="aim">Aim<a class="anchor" aria-label="anchor" href="#aim"></a>
</h4>
<p>To identify R packages and functions that can read, write or
interact/visualise data in the EML format</p>
<div class="section level5">
<h5 id="objectives">Objectives<a class="anchor" aria-label="anchor" href="#objectives"></a>
</h5>
<ol style="list-style-type: decimal">
<li>To review the available packages on CRAN and on GitHub that have
functions to interact with EML</li>
<li>To identify the main functions from these packages that allow
reading, writing and visualising data in EML format</li>
</ol>
</div>
</div>
</div>
<div class="section level3">
<h3 id="methods">Methods<a class="anchor" aria-label="anchor" href="#methods"></a>
</h3>
<p>We searched for R packages on CRAN (The Comprehensive R Archive
Network - <a href="https://cran.r-project.org/" class="external-link uri">https://cran.r-project.org/</a>) using the search function
which is provided by Google. We used the search term “Ecological
Metadata Language site:r-project.org”. We repeated the search for
“Ecological Metadata Language” on GitHub and identified repositories
which used the R language. We identified all packages returned through
our search that were on CRAN or on GitHub and that consisted of
functions that are aimed at interacting (either reading, writing or
viewing) with the Ecological Metadata Language (EML).</p>
</div>
<div class="section level3">
<h3 id="results">Results<a class="anchor" aria-label="anchor" href="#results"></a>
</h3>
<p>The search on Google (via CRAN) returned 25 hits. Seven of these were
R packages. The search on GitHub returned 15 repositories of which 6
were in the R language. Two of these returned errors when we attempted
to download them in RStudio using the devtools::install_github()
function, and two have been archived. We identified 9 R packages (6 from
CRAN and 3 from GitHub) that were active in the last two years and
explicitly stated that functions were used for interacting (either
reading, writing or viewing) Ecological Metadata Language (EML). Since
the original google search, several additional packages have become
available on GitHub and are not also included in the following list. One
of these, NPSdataverse is a wrapper package that will install and load
multiple EML-related packages including EML, EMLAssemblyline, EMLeditor,
DPchecker, and NPSutils. In addition, as of 2023-11-29, the package
rbefdata is no longer available via CRAN and is not being maintained by
on GitHub.</p>
<table class="table">
<colgroup>
<col width="1%">
<col width="98%">
</colgroup>
<thead><tr class="header">
<th align="left">Package</th>
<th align="left">Package_description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">EML</td>
<td align="left">Work with Ecological Metadata Language (‘EML’) files.
‘EML’ is a widely used metadata standard in the ecological and
environmental sciences, described in Jones et al. (2006), <a href="doi:10.1146/annurev.ecolsys.37.091305.110031" class="uri">doi:10.1146/annurev.ecolsys.37.091305.110031</a>.</td>
</tr>
<tr class="even">
<td align="left">emld</td>
<td align="left">This is a utility for transforming Ecological Metadata
Language (‘EML’) files into ‘JSON-LD’ and back into ‘EML.’ Doing so
creates a list-based representation of ‘EML’ in R, so that ‘EML’ data
can easily be manipulated using standard ‘R’ tools. This makes this
package an effective backend for other ‘R’-based tools working with
‘EML.’ By abstracting away the complexity of ‘XML’ Schema, developers
can build around native ‘R’ list objects and not have to worry about
satisfying many of the additional constraints of set by the schema (such
as element ordering, which is handled automatically). Additionally, the
‘JSON-LD’ representation enables the use of developer-friendly ‘JSON’
parsing and serialization that may facilitate the use of ‘EML’ in
contexts outside of ‘R,’ as well as the informatics-friendly
serializations such as ‘RDF’ and ‘SPARQL’ queries.</td>
</tr>
<tr class="odd">
<td align="left">RNeXML</td>
<td align="left">Provides access to phyloinformatic data in ‘NeXML’
format. The package should add new functionality to R such as the
possibility to manipulate ‘NeXML’ objects in more various and refined
way and compatibility with ‘ape’ objects.</td>
</tr>
<tr class="even">
<td align="left">datapack</td>
<td align="left">Provides a flexible container to transport and
manipulate complex sets of data. These data may consist of multiple data
files and associated meta data and ancillary files. Individual data
objects have associated system level meta data, and data files are
linked together using the OAI-ORE standard resource map which describes
the relationships between the files. The OAI- ORE standard is described
at <a href="https://www.openarchives.org/ore/" class="external-link uri">https://www.openarchives.org/ore/</a>. Data packages can be
serialized and transported as structured files that have been created
following the BagIt specification. The BagIt specification is described
at <a href="https://tools.ietf.org/html/draft-kunze-bagit-08" class="external-link uri">https://tools.ietf.org/html/draft-kunze-bagit-08</a>.</td>
</tr>
<tr class="odd">
<td align="left">geometa</td>
<td align="left">Provides facilities to handle reading and writing of
geographic metadata defined with OGC/ISO 19115, 11119 and 19110
geographic information metadata standards, and encoded using the ISO
19139 (XML) standard. It includes also a facility to check the validity
of ISO 19139 XML encoded metadata.</td>
</tr>
<tr class="even">
<td align="left">MetaEgress</td>
<td align="left">Functions to create Ecological Metadata Language (EML)
XML documents from metadata stored in a metadata database design by the
Long Term Ecological Research Network.</td>
</tr>
<tr class="odd">
<td align="left">EMLassemblyline</td>
<td align="left">For scientists and data managers to create high quality
EML metadata for dataset publication. EMLassemblyline is optimized for
automating recurring publications (timeseries or data derived from
timeseries sources) but works well for “one-off” publications,
especially through the MetaShARK interface. EMLassemblyline prioritizes
automated metadata extraction from data objects to minimize required
human effort and encourages EML best practices to make publications
Findable, Accessible, Interoperable, and Reusable.</td>
</tr>
<tr class="even">
<td align="left">pkEML</td>
<td align="left">Use this package to convert an existing corpus of
Ecological Metadata Language (EML) documents into table form, plus
identify and consolidate re-used metadata elements. Existing metadata
can then be imported or migrated into a relational database system with
re-usable metadata elements (with “primary keys” hence pkEML). This
package is developed with the intention to help established LTER sites
import their EML corpus into LTER-core-metabase. See <a href="https://github.com/lter/LTER-core-metabase" class="external-link uri">https://github.com/lter/LTER-core-metabase</a>. However
please use to whatever use you may find, and</td>
</tr>
<tr class="odd">
<td align="left">LivingNorwayR</td>
<td align="left">The package provides a workflow for creating a Darwin
Core standard-compliant data archive (“a data package”). This
facilitates FAIR (Findable, Accessible, Interoperable, Reusable; <a href="https://www.go-fair.org/fair-principles/" class="external-link uri">https://www.go-fair.org/fair-principles/</a>) data sharing
and uploading of Darwin Core archives (data packages) to repositories
such as GBIF. The Living Norway package also provides tools for the
processing and manipulation of metadata associated with Darwin Core
archives and for the import and export of metadata according to the EML
(Ecological Metadata Language; <a href="https://eml.ecoinformatics.org/" class="external-link uri">https://eml.ecoinformatics.org/</a>) standard.</td>
</tr>
<tr class="even">
<td align="left">NPSdataverse</td>
<td align="left">Loads a suite of R packages for creating and
manipulating data packages including interacting with DataStore.</td>
</tr>
<tr class="odd">
<td align="left">EMLeditor</td>
<td align="left">This package will be of most use to the U.S. National
Park Service data scientists and managers seeking to generate
EML-formatted metadata for datapackages. EML-formatted .xml files are
typically constructed using EDI’s EMLassemblyline package and then
imported as an R-object using the EML package. EMLeditor allows the user
to view the contents of the R object and add/edit aspects of metadata
crucial for publication in the U.S. National Park Service DataStore
repository. For instance, a user can view and edit a DOI, a link to a
DRR, Park Unit connections, information about Confidential Unclassified
Information (CUI), and more. EMLeditor allows the user to write a mockup
of a README.txt to preview what the README automatically generated by
DataStore upon upload will look like.</td>
</tr>
<tr class="even">
<td align="left">DPchecker</td>
<td align="left">Allows the user (and reviewer) to check a data package
and test whether it meets the congruence standards set forth by NPS for
upload to DataStore as a datapackage.</td>
</tr>
<tr class="odd">
<td align="left">NPSutils</td>
<td align="left">NPSutils is a collection of functions for interacting
with NPS DataStore repository.</td>
</tr>
</tbody>
</table>
<div class="section level5">
<h5 id="emld-package">emld package<a class="anchor" aria-label="anchor" href="#emld-package"></a>
</h5>
<p>The <em>emld</em> package <span class="citation">(Boettiger
2019)</span> is closely related to the EML package <span class="citation">(Boettiger and Jones 2021)</span> and its functions are
heavily relied upon in the latest version of <em>EML</em>.</p>
<p><em>emld</em> has an advantage over <em>EML</em> where there are
large, highly nested EML files. It can flatten EML in to common R
formats that can be manipulated in R.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example from https://github.com/ropensci/emld</span></span>
<span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/example.xml"</span>, package<span class="op">=</span><span class="st">"emld"</span><span class="op">)</span></span>
<span><span class="va">eml</span> <span class="op">&lt;-</span> <span class="fu">emld</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/emld/reference/as_emld.html" class="external-link">as_emld</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span></span>
<span><span class="va">eml</span><span class="op">$</span><span class="va">dataset</span><span class="op">$</span><span class="va">title</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Data from Cedar Creek LTER on productivity and species richness\n  for use in a workshop titled \"An Analysis of the Relationship between\n  Productivity and Diversity using Experimental Results from the Long-Term\n  Ecological Research Network\" held at NCEAS in September 1996."</span></span></code></pre>
<p><em>emld</em> objects are nested lists so to write EML in
<em>emld</em> you can create a list.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example from https://github.com/ropensci/emld</span></span>
<span></span>
<span><span class="va">me</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>individualName <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>givenName <span class="op">=</span> <span class="st">"Joe"</span>, surName <span class="op">=</span> <span class="st">"Bloggs"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">eml</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>dataset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>              title <span class="op">=</span> <span class="st">"The biggest fish in the sea"</span>,</span>
<span>              contact <span class="op">=</span> <span class="va">me</span>,</span>
<span>              creator <span class="op">=</span> <span class="va">me</span><span class="op">)</span>,</span>
<span>              system <span class="op">=</span> <span class="st">"doi"</span>,</span>
<span>              packageId <span class="op">=</span> <span class="st">"10.xxx"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ex.xml</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="st">"ex"</span>, fileext <span class="op">=</span> <span class="st">".xml"</span><span class="op">)</span> <span class="co"># use your preferred file path</span></span>
<span></span>
<span><span class="fu">as_xml</span><span class="op">(</span><span class="va">eml</span>, <span class="va">ex.xml</span><span class="op">)</span></span></code></pre></div>
<p><em>emld::as_xml()</em> reorders the list if necessary to ensure that
it matches the EML required format. You can use
<em>emld::eml_validate()</em> to check that the EML has been written
correctly and is a valid EML file.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">eml_validate</span><span class="op">(</span><span class="va">ex.xml</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span>
<span><span class="co">## attr(,"errors")</span></span>
<span><span class="co">## character(0)</span></span></code></pre>
</div>
<div class="section level5">
<h5 id="eml-package">EML package<a class="anchor" aria-label="anchor" href="#eml-package"></a>
</h5>
<p>The <em>EML</em> <span class="citation">(Boettiger and Jones
2021)</span> package uses <em>emld</em> <span class="citation">(Boettiger 2019)</span> as a basis for many of its
functions.</p>
<p>The read_eml() function reads eml files in to the R session.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"example.xml"</span>, package <span class="op">=</span> <span class="st">"emld"</span><span class="op">)</span></span>
<span><span class="va">eml</span> <span class="op">&lt;-</span> <span class="fu">EML</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/EML/reference/read_eml.html" class="external-link">read_eml</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span></span>
<span><span class="va">eml</span><span class="op">$</span><span class="va">dataset</span><span class="op">$</span><span class="va">title</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Data from Cedar Creek LTER on productivity and species richness\n  for use in a workshop titled \"An Analysis of the Relationship between\n  Productivity and Diversity using Experimental Results from the Long-Term\n  Ecological Research Network\" held at NCEAS in September 1996."</span></span></code></pre>
<p>Several functions (with the prefix <em>set_</em>) allow the user to
import attributes of the dataset file (e.g. the methods) as word or
Markdown documents and merge them in to a single EML file (see <a href="https://docs.ropensci.org/EML/" class="external-link uri">https://docs.ropensci.org/EML/</a> for a detailed
tutorial).</p>
<p><em>EML</em> has several functions (with the prefix <em>get_</em>)
which allow a user to extract elements of an XML file.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"tests"</span>, <span class="fu">emld</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/emld/reference/eml_version.html" class="external-link">eml_version</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>  <span class="st">"eml-datasetWithAttributelevelMethods.xml"</span>, package <span class="op">=</span> <span class="st">"emld"</span><span class="op">)</span></span>
<span><span class="va">eml</span> <span class="op">&lt;-</span> <span class="fu">EML</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/EML/reference/read_eml.html" class="external-link">read_eml</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span></span>
<span><span class="fu">EML</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/EML/reference/get_attributes.html" class="external-link">get_attributes</a></span><span class="op">(</span><span class="va">eml</span><span class="op">$</span><span class="va">dataset</span><span class="op">$</span><span class="va">dataTable</span><span class="op">$</span><span class="va">attributeList</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $attributes</span></span>
<span><span class="co">##    id...1 attributeName                    attributeLabel</span></span>
<span><span class="co">## 1   att.1           fld                             Field</span></span>
<span><span class="co">## 2   att.2          year                              year</span></span>
<span><span class="co">## 3   att.3            sr                  Species Richness</span></span>
<span><span class="co">## 4   att.4        pctcov                     percent cover</span></span>
<span><span class="co">## 5   att.5       avesr91 Average Species Richness for 1991</span></span>
<span><span class="co">## 6    &lt;NA&gt;       avesr92 Average Species Richness for 1992</span></span>
<span><span class="co">## 7    &lt;NA&gt;       avesr93 Average Species Richness for 1993</span></span>
<span><span class="co">## 8    &lt;NA&gt;       avesr94 Average Species Richness for 1994</span></span>
<span><span class="co">## 9    &lt;NA&gt;       avesr95 Average Species Richness for 1995</span></span>
<span><span class="co">## 10   &lt;NA&gt;       avesr96 Average Species Richness for 1996</span></span>
<span><span class="co">## 11   &lt;NA&gt;        MeanSR             mean species richness</span></span>
<span><span class="co">## 12 att.14          time                              Time</span></span>
<span><span class="co">##                                              attributeDefinition storageType</span></span>
<span><span class="co">## 1                   Field where the data was collected\n              string</span></span>
<span><span class="co">## 2                      The year the data was collected\n               gYear</span></span>
<span><span class="co">## 3                                       Species richness for CDR       float</span></span>
<span><span class="co">## 4                The percent ground cover on the field\n               float</span></span>
<span><span class="co">## 5   The average species richness for the field in 1991\n               float</span></span>
<span><span class="co">## 6   The average species richness for the field in 1992\n               float</span></span>
<span><span class="co">## 7   The average species richness for the field in 1993\n               float</span></span>
<span><span class="co">## 8   The average species richness for the field in 1994\n               float</span></span>
<span><span class="co">## 9   The average species richness for the field in 1995\n               float</span></span>
<span><span class="co">## 10  The average species richness for the field in 1996\n               float</span></span>
<span><span class="co">## 11         the mean species richness from 1991 to 1996\n               float</span></span>
<span><span class="co">## 12 The time of day for this observation, 24 hour clock\n                time</span></span>
<span><span class="co">##    id...6            definition measurementScale         domain formatString</span></span>
<span><span class="co">## 1    nd.1 Valid names of fields          nominal     textDomain         &lt;NA&gt;</span></span>
<span><span class="co">## 2    &lt;NA&gt;                  &lt;NA&gt;         dateTime dateTimeDomain         YYYY</span></span>
<span><span class="co">## 3    &lt;NA&gt;                  &lt;NA&gt;         interval  numericDomain         &lt;NA&gt;</span></span>
<span><span class="co">## 4    &lt;NA&gt;                  &lt;NA&gt;            ratio  numericDomain         &lt;NA&gt;</span></span>
<span><span class="co">## 5    &lt;NA&gt;                  &lt;NA&gt;            ratio  numericDomain         &lt;NA&gt;</span></span>
<span><span class="co">## 6    &lt;NA&gt;                  &lt;NA&gt;            ratio  numericDomain         &lt;NA&gt;</span></span>
<span><span class="co">## 7    &lt;NA&gt;                  &lt;NA&gt;            ratio  numericDomain         &lt;NA&gt;</span></span>
<span><span class="co">## 8    &lt;NA&gt;                  &lt;NA&gt;            ratio  numericDomain         &lt;NA&gt;</span></span>
<span><span class="co">## 9    &lt;NA&gt;                  &lt;NA&gt;            ratio  numericDomain         &lt;NA&gt;</span></span>
<span><span class="co">## 10   &lt;NA&gt;                  &lt;NA&gt;            ratio  numericDomain         &lt;NA&gt;</span></span>
<span><span class="co">## 11   &lt;NA&gt;                  &lt;NA&gt;            ratio  numericDomain         &lt;NA&gt;</span></span>
<span><span class="co">## 12   &lt;NA&gt;                  &lt;NA&gt;         dateTime dateTimeDomain   hh:mm:ss.s</span></span>
<span><span class="co">##    dateTimePrecision id...8    minimum          unit precision numberType</span></span>
<span><span class="co">## 1               &lt;NA&gt;   &lt;NA&gt;       &lt;NA&gt;          &lt;NA&gt;      &lt;NA&gt;       &lt;NA&gt;</span></span>
<span><span class="co">## 2                  1   dd.2       1944          &lt;NA&gt;      &lt;NA&gt;       &lt;NA&gt;</span></span>
<span><span class="co">## 3               &lt;NA&gt;   nd.3          0 dimensionless       0.5       real</span></span>
<span><span class="co">## 4               &lt;NA&gt;   nd.4          0 dimensionless       0.1       real</span></span>
<span><span class="co">## 5               &lt;NA&gt;   nd.5          0 dimensionless       0.1       real</span></span>
<span><span class="co">## 6               &lt;NA&gt;   &lt;NA&gt;       &lt;NA&gt; dimensionless       0.1       &lt;NA&gt;</span></span>
<span><span class="co">## 7               &lt;NA&gt;   &lt;NA&gt;       &lt;NA&gt; dimensionless       0.1       &lt;NA&gt;</span></span>
<span><span class="co">## 8               &lt;NA&gt;   &lt;NA&gt;       &lt;NA&gt; dimensionless       0.1       &lt;NA&gt;</span></span>
<span><span class="co">## 9               &lt;NA&gt;   &lt;NA&gt;       &lt;NA&gt; dimensionless       0.1       &lt;NA&gt;</span></span>
<span><span class="co">## 10              &lt;NA&gt;   &lt;NA&gt;       &lt;NA&gt; dimensionless       0.1       &lt;NA&gt;</span></span>
<span><span class="co">## 11              &lt;NA&gt;   &lt;NA&gt;       &lt;NA&gt; dimensionless       0.1       &lt;NA&gt;</span></span>
<span><span class="co">## 12               0.1   dd.3 15:00:00.0          &lt;NA&gt;      &lt;NA&gt;       &lt;NA&gt;</span></span>
<span><span class="co">##    exclusive...10 exclusive...12    maximum     id exclusive...9 exclusive...11</span></span>
<span><span class="co">## 1            &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;          &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">## 2            &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;          &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">## 3            &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;          &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">## 4            true           true        100   &lt;NA&gt;          &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">## 5            &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;          &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">## 6            &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;  att.6          &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">## 7            &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;  att.7          &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">## 8            &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;  att.8          &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">## 9            &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;  att.9          &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">## 10           &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt; att.10          &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">## 11           &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt; att.11          &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">## 12           &lt;NA&gt;           &lt;NA&gt; 19:00:00.0   &lt;NA&gt;          true           true</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $factors</span></span>
<span><span class="co">## NULL</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="rnexml-package">RNeXML package<a class="anchor" aria-label="anchor" href="#rnexml-package"></a>
</h4>
<p><em>RNeXML</em> <span class="citation">(Boettiger et al. 2016)</span>
is a package for reading and writing phylogenetic, character and trait
metadata focussed on taxonomy. The package stands parallel to
<em>EML</em> and <em>emd</em> using similar functions but is focused on
a different (but related) standard NeXML rather than EML per se.</p>
</div>
<div class="section level4">
<h4 id="datapack-package">datapack package<a class="anchor" aria-label="anchor" href="#datapack-package"></a>
</h4>
<p>The <em>datapack</em> package <span class="citation">(M. B. Jones and
Slaughter 2020)</span> provides functions for collating multiple data
and metadata objects of different types into a bundle that can be
transported and loaded using a single composite file. It is primarily
meant as a container to bundle together files for transport to or from
DataONE data repositories. Metadata in the form of EML can be attached
to the databundle.</p>
<pre><code><span><span class="co">## Members:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## filename       format    mediaType  size     identifier          modified local </span></span>
<span><span class="co">## sample-eml.xml eml...1.0 NA         5990     urn:uuid...18e46751 n        y     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Package identifier: NA</span></span>
<span><span class="co">## RightsHolder: NA</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## This package does not contain any provenance relationships.</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="rbefdata-package">rbefdata package<a class="anchor" aria-label="anchor" href="#rbefdata-package"></a>
</h4>
<p><strong>This package does not appear to be available on CRAN or
GitHub as of 2023-11-29</strong></p>
<p>Similarly to the <em>datapack</em> package <em>rbefdata</em> <span class="citation">(Pfaff, Nadrowski, and Man 2013)</span> links to the
BEF data portal (<a href="https://fundiv.befdata.biow.uni-leipzig.de/" class="external-link uri">https://fundiv.befdata.biow.uni-leipzig.de/</a>). The
function <em>rbefdata::bef.portal.get.metadata()</em> extracts the
metadata from a file that the user has downloaded from the BEF data
portal.</p>
</div>
<div class="section level4">
<h4 id="geometa-package">geometa package<a class="anchor" aria-label="anchor" href="#geometa-package"></a>
</h4>
<p><em>geometa</em> <span class="citation">(Blondel 2021)</span>
provides functions for reading and writing geographic metadata. It
suggests <em>EML</em> and <em>emld</em>. It can convert metadata to and
from EML using the <em>geometa::convert_metadata()</em> function.</p>
</div>
<div class="section level4">
<h4 id="metaegress-package">MetaEgress package<a class="anchor" aria-label="anchor" href="#metaegress-package"></a>
</h4>
<p><em>MetaEgress</em> <span class="citation">(Nguyen and Kui
2021)</span> is designed to create EML for Long Term Ecological Research
metabase ( <a href="https://github.com/lter/LTER-core-metabase" class="external-link uri">https://github.com/lter/LTER-core-metabase</a>). Validation
of EML is done through <em>emld::eml_validate()</em></p>
</div>
<div class="section level4">
<h4 id="emlassemblyline-package">EMLassemblyline package<a class="anchor" aria-label="anchor" href="#emlassemblyline-package"></a>
</h4>
<p><em>EMLassemblyline</em> <span class="citation">(Smith 2021)</span>
is a metadata builder that has functions to autoextract metadata. The
package is centered around a “data package” that is a collection of data
objects and metadata. The templating functions allow the development of
metadata templates that the user can add information to.
<em>EMLassemblyline</em> also has a “living data” set of functions that
allow ongoing data collection to be published at regular periods. See <a href="https://ediorg.github.io/EMLassemblyline/index.html" class="external-link uri">https://ediorg.github.io/EMLassemblyline/index.html</a> for
examples of the workflow.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/EDIorg/EMLassemblyline" class="external-link">EMLassemblyline</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ediorg.github.io/EMLassemblyline/reference/make_eml.html" class="external-link">make_eml</a></span><span class="op">(</span></span>
<span>  path <span class="op">=</span> <span class="st">"./metadata_templates"</span>,</span>
<span>  data.path <span class="op">=</span> <span class="st">"./data_objects"</span>,</span>
<span>  eml.path <span class="op">=</span> <span class="st">"./eml"</span>,</span>
<span>  dataset.title <span class="op">=</span> <span class="st">"Sphagnum and Vascular Plant Decomposition under Increasing Nitrogen Additions"</span>,</span>
<span>  temporal.coverage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"2014-05-01"</span>, <span class="st">"2015-10-31"</span><span class="op">)</span>,</span>
<span>  maintenance.description <span class="op">=</span> <span class="st">"Completed: No updates to these data are expected"</span>,</span>
<span>  data.table <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"decomp.csv"</span>, <span class="st">"nitrogen.csv"</span><span class="op">)</span>,</span>
<span>  data.table.description <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Decomposition data"</span>, <span class="st">"Nitrogen data"</span><span class="op">)</span>,</span>
<span>  other.entity <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ancillary_data.zip"</span>, <span class="st">"processing_and_analysis.R"</span><span class="op">)</span>,</span>
<span>  other.entity.description <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Ancillary data"</span>, <span class="st">"Data processing and analysis script"</span><span class="op">)</span>,</span>
<span>  user.id <span class="op">=</span> <span class="st">"myid"</span>,</span>
<span>  user.domain <span class="op">=</span> <span class="st">"EDI"</span>,</span>
<span>  package.id <span class="op">=</span> <span class="st">"edi.260.1"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="metainbase-pkeml-package">MetaInbase (pkEML) package<a class="anchor" aria-label="anchor" href="#metainbase-pkeml-package"></a>
</h4>
<p><em>MetaInbase</em> <span class="citation">(c)) 2021)</span> has
functions to convert EML to tables. It appears to be in an early stage
of development. Renamed to pkEML.</p>
</div>
<div class="section level4">
<h4 id="metashark-package">MetaShARK package<a class="anchor" aria-label="anchor" href="#metashark-package"></a>
</h4>
<p><em>MetaShARK</em> (Metadata Shiny Automated Resource &amp;
Knowledge; <a href="https://github.com/earnaud/MetaShARK-v2" class="external-link uri">https://github.com/earnaud/MetaShARK-v2</a>) is a R shiny
app allowing the user to get information about EML and to fill in
metadata for datasets according to this standard. This package has
developed a “user-friendly” Shiny App for researchers not familiar with
EML standards.</p>
</div>
<div class="section level4">
<h4 id="livingnorwayr">LivingNorwayR<a class="anchor" aria-label="anchor" href="#livingnorwayr"></a>
</h4>
<p><em>LivingNorwayR</em> addresses the EML metadata in a slightly
different way to the other packages. It uses RMarkdown with specific
tagging functions in order to “knit” together the EML document. The
package is focused on creating Darwin Core compliant data archives
(“data packages”) that can be stored locally or shared with, for
example, GBIF. A worked example is available as a <a href="https://raw.githubusercontent.com/LivingNorway/LivingNorwayR/master/vignettes/LNWorkshopExample_TOV-E_Metadata.rmd" class="external-link">vignette</a>.</p>
</div>
<div class="section level4">
<h4 id="npsdataverse">NPSdataverse<a class="anchor" aria-label="anchor" href="#npsdataverse"></a>
</h4>
<p><em>NPSdataverse</em> is a wrapper package that will install a suite
of packages for generating, editing, checking, up/downloading (to/from
the NPS <a href="https://irma.nps.gov/DataStore/" class="external-link">DataStore</a>) and
accessing EML. <a href="https://nationalparkservice.github.io/NPSdataverse/" class="external-link">NPSdataverse</a>
will install and load the following packages: EML, EMLassemblyline,
QCkit, EMLeditor, DPchecker, NPSutils. Of these, QCkit is the only
package that does not involve EML in one way or another.</p>
</div>
<div class="section level4">
<h4 id="emleditor">EMLeditor<a class="anchor" aria-label="anchor" href="#emleditor"></a>
</h4>
<p><em>EMLeditor</em> is a package aimed at editing EML. <a href="https://nationalparkservice.github.io/EMLeditor/" class="external-link">EMLeditor</a>
package contains a number of “get_” and “set_” class functions for
retrieving and editing EML objects in R. Installing EMLeditor will also
install a sample .rmd script accessible via Rstudio for generating EML
via a combination of EML, EMLassemblyline, and EMLeditor. One major goal
of EMLeditor is to make editing EML fast and easy without needing to
call the make_eml() function from EMLassemblyline. Another goal is to
add EML elements that may be specific to National Park Staff,
collaborators, or partners. The package will also authenticated users to
generate references on the NPS DataStore and upload data sets/metadata
to DataStore.</p>
</div>
<div class="section level4">
<h4 id="dpchecker">DPchecker<a class="anchor" aria-label="anchor" href="#dpchecker"></a>
</h4>
<p><em>DPchecker</em> is a package that aimed at checking whether a data
package - consisting of several flat files in .csv format and a single
EML file (*_metadata.xml) is ready to be uploaded to the NPS DataStore.
Although many of the checks will be NPS specific, the <a href="https://nationalparkservice.github.io/DPchecker/" class="external-link">DPchecker</a>
also performs a number of more general checks for internal consitency in
the EML file as well as congruence between the EML file and the data
files.</p>
</div>
<div class="section level4">
<h4 id="npsutils">NPSutils<a class="anchor" aria-label="anchor" href="#npsutils"></a>
</h4>
<p><em>NPSutils</em> is primarily a public-facing package aimed at
accessing data packages on the NPS DataStore. <a href="https://nationalparkservice.github.io/NPSutils/" class="external-link">NPSutils</a>
contains functions to download data packages (.csv files and an EML
file), load the data and metadata in to R, and load the data and EML
metadata into dashboard visualizers such as Power BI.</p>
</div>
</div>
<div class="section level3">
<h3 id="relationship-between-packages">Relationship between packages<a class="anchor" aria-label="anchor" href="#relationship-between-packages"></a>
</h3>
<p>By looking at the dependency relationship between the packages we can
get an understanding of how they are structured and how similar each
package is. From the figure and table below you can see that
<em>jsonlite</em> and XML2 are core packages that underpin many of the
packages that deal with EML in R. <em>jsonlite</em> helps to convert the
list-like structure of EML in to different R classes. XML2 allows the
user to read in XML files (EML is a type of XML) to R as a list and to
also write EML as an XML structured file. Three of the packages use the
<em>emld</em> package as a basis (most use the validation function) and
from the network plot you can see that all the packages except
<em>EMLassemblyline</em> are quite closely related in terms of their
underlying dependencies. A large number of the dependencies of
<em>EMLassemblyline</em> are not shared by the other packages.</p>
<div id="htmlwidget-d2753834350e97ef28e0" style="width:100%;height:432.632880098888px;" class="visNetwork html-widget "></div>
<script type="application/json" data-for="htmlwidget-d2753834350e97ef28e0">{"x":{"nodes":{"id":["EML","emld","RNeXML","datapack","geometa","MetaEgress","EMLassemblyline","NPSdataverse","EMLeditor","DPchecker","NPSutils","R6","Rcpp","V8","base64enc","bslib","cachem","cli","curl","digest","dplyr","ellipsis","evaluate","fansi","fastmap","fontawesome","fs","generics","glue","grDevices","graphics","highr","htmltools","jqr","jquerylib","jsonld","jsonlite","knitr","lazyeval","lifecycle","magrittr","memoise","methods","mime","pillar","pkgconfig","rappdirs","rlang","rmarkdown","sass","stats","stringi","stringr","tibble","tidyselect","tinytex","tools","utf8","utils","uuid","vctrs","withr","xfun","xml2","yaml","XML","ape","askpass","compiler","grid","httr","lattice","nlme","openssl","parallel","plyr","purrr","reshape2","sys","tidyr","brew","callr","commonmark","crayon","desc","pkgbuild","pkgload","prettyunits","processx","ps","redland","roxygen2","rprojroot","zip","assertthat","bit","bit64","clipr","filelock","hms","keyring","readr","sodium","tzdb","vroom","DBI","EDIutils","RPostgres","blob","lubridate","timechange","Formula","Hmisc","MASS","Matrix","NCmisc","RColorBrewer","WikidataQueryServiceR","WikidataR","WikipediR","backports","bold","checkmate","cluster","codetools","colorspace","conditionz","covr","crul","data.table","farver","fastmatch","foreach","foreign","ggplot2","gridExtra","gtable","htmlTable","htmlwidgets","httpcode","igraph","isoband","iterators","labeling","mgcv","munsell","natserv","nnet","pbapply","phangorn","progress","quadprog","ratelimitr","reader","rentrez","rex","ritis","rncl","rotl","rpart","rredlist","rstudioapi","scales","solrium","splines","taxize","taxonomyCleanr","triebeard","urltools","viridis","viridisLite","wikitaxa","worrms","zoo","ISOcodes","KernSmooth","ODataQuery","QCkit","arcticdatautils","base","class","classInt","crosstalk","dataone","e1071","gdata","gtools","here","huxtable","leaflet","leaflet.providers","parsedate","pins","png","proxy","raster","rdflib","remotes","s2","sf","sp","stargazer","terra","units","whisker","wk"],"label":["EML","emld","RNeXML","datapack","geometa","MetaEgress","EMLassemblyline","NPSdataverse","EMLeditor","DPchecker","NPSutils","R6","Rcpp","V8","base64enc","bslib","cachem","cli","curl","digest","dplyr","ellipsis","evaluate","fansi","fastmap","fontawesome","fs","generics","glue","grDevices","graphics","highr","htmltools","jqr","jquerylib","jsonld","jsonlite","knitr","lazyeval","lifecycle","magrittr","memoise","methods","mime","pillar","pkgconfig","rappdirs","rlang","rmarkdown","sass","stats","stringi","stringr","tibble","tidyselect","tinytex","tools","utf8","utils","uuid","vctrs","withr","xfun","xml2","yaml","XML","ape","askpass","compiler","grid","httr","lattice","nlme","openssl","parallel","plyr","purrr","reshape2","sys","tidyr","brew","callr","commonmark","crayon","desc","pkgbuild","pkgload","prettyunits","processx","ps","redland","roxygen2","rprojroot","zip","assertthat","bit","bit64","clipr","filelock","hms","keyring","readr","sodium","tzdb","vroom","DBI","EDIutils","RPostgres","blob","lubridate","timechange","Formula","Hmisc","MASS","Matrix","NCmisc","RColorBrewer","WikidataQueryServiceR","WikidataR","WikipediR","backports","bold","checkmate","cluster","codetools","colorspace","conditionz","covr","crul","data.table","farver","fastmatch","foreach","foreign","ggplot2","gridExtra","gtable","htmlTable","htmlwidgets","httpcode","igraph","isoband","iterators","labeling","mgcv","munsell","natserv","nnet","pbapply","phangorn","progress","quadprog","ratelimitr","reader","rentrez","rex","ritis","rncl","rotl","rpart","rredlist","rstudioapi","scales","solrium","splines","taxize","taxonomyCleanr","triebeard","urltools","viridis","viridisLite","wikitaxa","worrms","zoo","ISOcodes","KernSmooth","ODataQuery","QCkit","arcticdatautils","base","class","classInt","crosstalk","dataone","e1071","gdata","gtools","here","huxtable","leaflet","leaflet.providers","parsedate","pins","png","proxy","raster","rdflib","remotes","s2","sf","sp","stargazer","terra","units","whisker","wk"]},"edges":{"from":["R6","Rcpp","V8","base64enc","bslib","cachem","cli","curl","digest","dplyr","ellipsis","emld","evaluate","fansi","fastmap","fontawesome","fs","generics","glue","grDevices","graphics","highr","htmltools","jqr","jquerylib","jsonld","jsonlite","knitr","lazyeval","lifecycle","magrittr","memoise","methods","mime","pillar","pkgconfig","rappdirs","rlang","rmarkdown","sass","stats","stringi","stringr","tibble","tidyselect","tinytex","tools","utf8","utils","uuid","vctrs","withr","xfun","xml2","yaml","Rcpp","V8","curl","grDevices","graphics","jsonld","jsonlite","methods","stats","utils","xml2","yaml","R6","Rcpp","XML","ape","askpass","cli","compiler","curl","digest","dplyr","fansi","generics","glue","grDevices","graphics","grid","httr","jsonlite","lattice","lifecycle","magrittr","methods","mime","nlme","openssl","parallel","pillar","pkgconfig","plyr","purrr","reshape2","rlang","stats","stringi","stringr","sys","tibble","tidyr","tidyselect","tools","utf8","utils","uuid","vctrs","withr","xml2","R6","XML","brew","callr","cli","commonmark","crayon","desc","digest","evaluate","fs","glue","grDevices","graphics","highr","knitr","lifecycle","magrittr","methods","pkgbuild","pkgload","prettyunits","processx","ps","purrr","redland","rlang","roxygen2","rprojroot","stats","stringi","stringr","tools","utils","uuid","vctrs","withr","xfun","xml2","yaml","zip","R6","XML","askpass","assertthat","bit","bit64","cli","clipr","crayon","curl","fansi","filelock","glue","grDevices","graphics","hms","httr","jsonlite","keyring","lifecycle","magrittr","methods","mime","openssl","pillar","pkgconfig","rappdirs","readr","rlang","sodium","stats","sys","tibble","tidyselect","tools","tzdb","utf8","utils","vctrs","vroom","withr","yaml","DBI","EDIutils","EML","R6","RPostgres","Rcpp","V8","askpass","base64enc","bit","bit64","blob","bslib","cachem","cli","clipr","crayon","curl","digest","dplyr","ellipsis","emld","evaluate","fansi","fastmap","fontawesome","fs","generics","glue","grDevices","graphics","highr","hms","htmltools","httr","jqr","jquerylib","jsonld","jsonlite","knitr","lazyeval","lifecycle","lubridate","magrittr","memoise","methods","mime","openssl","pillar","pkgconfig","rappdirs","readr","rlang","rmarkdown","sass","stats","stringi","stringr","sys","tibble","tidyselect","timechange","tinytex","tools","tzdb","utf8","utils","uuid","vctrs","vroom","withr","xfun","xml2","yaml","EML","Formula","Hmisc","MASS","Matrix","NCmisc","R6","RColorBrewer","Rcpp","V8","WikidataQueryServiceR","WikidataR","WikipediR","XML","ape","askpass","assertthat","backports","base64enc","bit","bit64","bold","bslib","cachem","checkmate","cli","clipr","cluster","codetools","colorspace","compiler","conditionz","covr","crayon","crul","curl","data.table","digest","dplyr","ellipsis","emld","evaluate","fansi","farver","fastmap","fastmatch","fontawesome","foreach","foreign","fs","generics","ggplot2","glue","grDevices","graphics","grid","gridExtra","gtable","highr","hms","htmlTable","htmltools","htmlwidgets","httpcode","httr","igraph","isoband","iterators","jqr","jquerylib","jsonld","jsonlite","knitr","labeling","lattice","lazyeval","lifecycle","magrittr","memoise","methods","mgcv","mime","munsell","natserv","nlme","nnet","openssl","parallel","pbapply","phangorn","pillar","pkgconfig","plyr","prettyunits","progress","purrr","quadprog","rappdirs","ratelimitr","reader","readr","rentrez","rex","ritis","rlang","rmarkdown","rncl","rotl","rpart","rredlist","rstudioapi","sass","scales","solrium","splines","stats","stringi","stringr","sys","taxize","taxonomyCleanr","tibble","tidyr","tidyselect","tinytex","tools","triebeard","tzdb","urltools","utf8","utils","uuid","vctrs","viridis","viridisLite","vroom","wikitaxa","withr","worrms","xfun","xml2","yaml","zoo","DBI","DPchecker","EML","EMLassemblyline","EMLeditor","Formula","Hmisc","ISOcodes","KernSmooth","MASS","Matrix","NCmisc","NPSutils","ODataQuery","QCkit","R6","RColorBrewer","Rcpp","V8","WikidataQueryServiceR","WikidataR","WikipediR","XML","ape","arcticdatautils","askpass","assertthat","backports","base","base64enc","bit","bit64","bold","brew","bslib","cachem","callr","checkmate","class","classInt","cli","clipr","cluster","codetools","colorspace","commonmark","compiler","conditionz","covr","crayon","crosstalk","crul","curl","data.table","dataone","datapack","desc","digest","dplyr","e1071","ellipsis","emld","evaluate","fansi","farver","fastmap","fastmatch","filelock","fontawesome","foreach","foreign","fs","gdata","generics","ggplot2","glue","grDevices","graphics","grid","gridExtra","gtable","gtools","here","highr","hms","htmlTable","htmltools","htmlwidgets","httpcode","httr","huxtable","igraph","isoband","iterators","jqr","jquerylib","jsonld","jsonlite","knitr","labeling","lattice","lazyeval","leaflet","leaflet.providers","lifecycle","lubridate","magrittr","memoise","methods","mgcv","mime","munsell","natserv","nlme","nnet","openssl","parallel","parsedate","pbapply","phangorn","pillar","pins","pkgbuild","pkgconfig","pkgload","plyr","png","prettyunits","processx","progress","proxy","ps","purrr","quadprog","rappdirs","raster","ratelimitr","rdflib","reader","readr","redland","remotes","rentrez","rex","ritis","rlang","rmarkdown","rncl","rotl","roxygen2","rpart","rprojroot","rredlist","rstudioapi","s2","sass","scales","sf","solrium","sp","splines","stargazer","stats","stringi","stringr","sys","taxize","taxonomyCleanr","terra","tibble","tidyr","tidyselect","timechange","tinytex","tools","triebeard","tzdb","units","urltools","utf8","utils","uuid","vctrs","viridis","viridisLite","vroom","whisker","wikitaxa","withr","wk","worrms","xfun","xml2","yaml","zip","zoo","DBI","DPchecker","EML","ISOcodes","KernSmooth","MASS","R6","Rcpp","V8","XML","arcticdatautils","askpass","base64enc","bit","bit64","brew","bslib","cachem","callr","class","classInt","cli","clipr","commonmark","crayon","curl","dataone","datapack","desc","digest","dplyr","e1071","ellipsis","emld","evaluate","fansi","fastmap","filelock","fontawesome","fs","gdata","generics","glue","grDevices","graphics","grid","gtools","here","highr","hms","htmltools","httr","jqr","jquerylib","jsonld","jsonlite","knitr","lazyeval","lifecycle","lubridate","magrittr","memoise","methods","mime","openssl","parsedate","pillar","pins","pkgbuild","pkgconfig","pkgload","plyr","prettyunits","processx","proxy","ps","purrr","rappdirs","rdflib","readr","redland","rlang","rmarkdown","roxygen2","rprojroot","s2","sass","sf","stargazer","stats","stringi","stringr","sys","tibble","tidyr","tidyselect","timechange","tinytex","tools","tzdb","units","utf8","utils","uuid","vctrs","vroom","whisker","withr","wk","xfun","xml2","yaml","zip","EML","R6","Rcpp","V8","XML","arcticdatautils","askpass","base64enc","bit","bit64","brew","bslib","cachem","callr","cli","clipr","commonmark","crayon","curl","dataone","datapack","desc","digest","dplyr","ellipsis","emld","evaluate","fansi","fastmap","filelock","fontawesome","fs","generics","glue","grDevices","graphics","highr","hms","htmltools","httr","jqr","jquerylib","jsonld","jsonlite","knitr","lazyeval","lifecycle","lubridate","magrittr","memoise","methods","mime","openssl","parsedate","pillar","pins","pkgbuild","pkgconfig","pkgload","plyr","prettyunits","processx","ps","purrr","rappdirs","rdflib","readr","redland","rlang","rmarkdown","roxygen2","rprojroot","sass","stats","stringi","stringr","sys","tibble","tidyr","tidyselect","timechange","tinytex","tools","tzdb","utf8","utils","uuid","vctrs","vroom","whisker","withr","xfun","xml2","yaml","zip","DBI","DPchecker","EML","EMLeditor","ISOcodes","KernSmooth","MASS","Matrix","R6","RColorBrewer","Rcpp","V8","XML","arcticdatautils","askpass","base64enc","bit","bit64","brew","bslib","cachem","callr","class","classInt","cli","clipr","colorspace","commonmark","crayon","crosstalk","curl","dataone","datapack","desc","digest","dplyr","e1071","ellipsis","emld","evaluate","fansi","farver","fastmap","filelock","fontawesome","fs","gdata","generics","ggplot2","glue","grDevices","graphics","grid","gridExtra","gtable","gtools","here","highr","hms","htmltools","htmlwidgets","httr","isoband","jqr","jquerylib","jsonld","jsonlite","knitr","labeling","lattice","lazyeval","leaflet","leaflet.providers","lifecycle","lubridate","magrittr","memoise","methods","mgcv","mime","munsell","nlme","openssl","parsedate","pillar","pins","pkgbuild","pkgconfig","pkgload","plyr","png","prettyunits","processx","proxy","ps","purrr","rappdirs","raster","rdflib","readr","redland","rlang","rmarkdown","roxygen2","rprojroot","s2","sass","scales","sf","sp","splines","stargazer","stats","stringi","stringr","sys","terra","tibble","tidyr","tidyselect","timechange","tinytex","tools","tzdb","units","utf8","utils","uuid","vctrs","viridis","viridisLite","vroom","whisker","withr","wk","xfun","xml2","yaml","zip"],"to":["EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","emld","emld","emld","emld","emld","emld","emld","emld","emld","emld","emld","emld","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","NPSdataverse","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","EMLeditor","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","DPchecker","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils","NPSutils"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot"},"manipulation":{"enabled":false},"interaction":{"dragNodes":false,"dragView":false,"hover":true,"navigationButtons":true,"zoomSpeed":1},"edges":{"arrows":{"to":{"enabled":true,"scaleFactor":2}},"color":{"color":"lightblue","highlight":"red"}},"physics":{"solver":"forceAtlas2Based","forceAtlas2Based":{"gravitationalConstant":-300}}},"groups":null,"width":"100%","height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","tooltipStay":300,"tooltipStyle":"position: fixed;visibility:hidden;padding: 5px;white-space: nowrap;font-family: verdana;font-size:14px;font-color:#000000;background-color: #f5f4ed;-moz-border-radius: 3px;-webkit-border-radius: 3px;border-radius: 3px;border: 1px solid #808074;box-shadow: 3px 3px 10px rgba(0, 0, 0, 0.2);"},"evals":[],"jsHooks":[]}</script><div class="section level4">
<h4 id="which-package-to-use">Which package to use?<a class="anchor" aria-label="anchor" href="#which-package-to-use"></a>
</h4>
<p>Many of the packages we have found are focused on single data
repositories. These packages might be useful for these specific tasks,
however the main packages for creating EML are {emld} and {EML} as these
are the basis to many of the functions used in the other packages. The
{LivingNorwayR} package uses RMarkdown as a tool to create EML metadata
and might be preferable to those with some experience with RMarkdown
documents.</p>
</div>
</div>
<div class="section level3">
<h3 id="places-to-go-for-help">Places to go for help<a class="anchor" aria-label="anchor" href="#places-to-go-for-help"></a>
</h3>
<ul>
<li><a href="https://environmentaldatainitiative.org/five-phases-of-data-publishing/phase-3/make-eml-with-r/" class="external-link uri">https://environmentaldatainitiative.org/five-phases-of-data-publishing/phase-3/make-eml-with-r/</a></li>
<li><a href="https://cran.r-project.org/web/packages/EML/vignettes/creating-EML.html" class="external-link uri">https://cran.r-project.org/web/packages/EML/vignettes/creating-EML.html</a></li>
<li><a href="https://cran.r-project.org/web/packages/emld/vignettes/tutorial.html" class="external-link uri">https://cran.r-project.org/web/packages/emld/vignettes/tutorial.html</a></li>
<li><a href="https://www.neonscience.org/dc-metadata-importance-eml-r" class="external-link uri">https://www.neonscience.org/dc-metadata-importance-eml-r</a></li>
<li><a href="https://ropensci.org/blog/2017/08/01/emldown/" class="external-link uri">https://ropensci.org/blog/2017/08/01/emldown/</a></li>
</ul>
<div class="section level5">
<h5 id="contributors">Contributors<a class="anchor" aria-label="anchor" href="#contributors"></a>
</h5>
<p><a href="https://github.com/DrMattG" class="external-link">Matt Grainger</a></p>
<p><a href="https://github.com/RobLBaker" class="external-link">Rob Baker</a></p>
</div>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-geometa" class="csl-entry">
Blondel, Emmanuel. 2021. <em>Geometa: Tools for Reading and Writing
ISO/OGC Geographic Metadata</em>. <a href="https://CRAN.R-project.org/package=geometa" class="external-link">https://CRAN.R-project.org/package=geometa</a>.
</div>
<div id="ref-emld" class="csl-entry">
Boettiger, Carl. 2019. <span>“Ecological Metadata as Linked Data.
Journal of Open Source Software.”</span> <em>The Journal of Open Source
Software</em> 4 (34): 1276. <a href="https://doi.org/10.21105/joss.01276" class="external-link">https://doi.org/10.21105/joss.01276</a>.
</div>
<div id="ref-RNeXML" class="csl-entry">
Boettiger, Carl, Scott Chamberlain, Rutger Vos, and Hilmar Lapp. 2016.
<span>“<span>RNeXML</span>: <span>A</span> Package for Reading and
Writing Richly Annotated Phylogenetic, Character, and Trait Data in
<span>R</span>.”</span> <em>Methods in Ecology and Evolution</em> 7:
352–57. <a href="https://doi.org/10.1111/2041-210X.12469" class="external-link">https://doi.org/10.1111/2041-210X.12469</a>.
</div>
<div id="ref-EML" class="csl-entry">
Boettiger, Carl, and Matthew B. Jones. 2021. <em>EML: Read and Write
Ecological Metadata Language Files</em>. <a href="https://CRAN.R-project.org/package=EML" class="external-link">https://CRAN.R-project.org/package=EML</a>.
</div>
<div id="ref-MetaInbase" class="csl-entry">
c)). 2021. <em>MetaInbase: Convert Corpus of Ecological Metadata
Language (EML) Documents to Table Form</em>.
</div>
<div id="ref-datapack" class="csl-entry">
Jones, Matthew B., and Peter Slaughter. 2020. <em><span class="nocase">datapack</span>: A Flexible Container to Transport and
Manipulate Data and Associated Resources</em>. <a href="https://doi.org/10.5063/F1QV3JGM" class="external-link">https://doi.org/10.5063/F1QV3JGM</a>.
</div>
<div id="ref-EML_2019" class="csl-entry">
Jones, Matthew, Margaret O’Brien, Bryce Mecum, Carl Boettiger, Mark
Schildhauer, Mitchell Maier, Timothy Whiteaker, Stevan Earl, and Steven
Chong. 2019. <span>“Ecological Metadata Language Version 2.2.0.”</span>
<a href="https://doi.org/10.5063/f11834t2" class="external-link">https://doi.org/10.5063/f11834t2</a>.
</div>
<div id="ref-mccartney2002using" class="csl-entry">
McCartney, P, and M Jones. 2002. <span>“Using XML-Encoded Metadata as a
Basis for Advanced Information Systems for Ecological Research.”</span>
In <em>Proc. 6th World Multiconference Systemics, Cybernetics and
Informatics</em>, 7:379–84.
</div>
<div id="ref-MetaEgress" class="csl-entry">
Nguyen, An T., and Li Kui. 2021. <em>MetaEgress: Create Ecological
Metadata Language from LTER Core Metabase</em>.
</div>
<div id="ref-rbefdata" class="csl-entry">
Pfaff, Claas-Thido, Karin Nadrowski, and Xingxing Man. 2013.
<em>Rbefdata: BEFdata r Package</em>. <a href="https://CRAN.R-project.org/package=rbefdata" class="external-link">https://CRAN.R-project.org/package=rbefdata</a>.
</div>
<div id="ref-EMLassemblyline" class="csl-entry">
Smith, Colin. 2021. <em>EMLassemblyline: A Tool Kit for Building EML
Metadata Workflows</em>. <a href="https://github.com/EDIorg/EMLassemblyline" class="external-link">https://github.com/EDIorg/EMLassemblyline</a>.
</div>
</div>
<div style="page-break-after: always;"></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Joseph Chipperfield, Matthew Grainger, Erlend Nilsen, Norsk institutt for naturforskning (NINA).</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
