<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>EML R package overview • LivingNorwayR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="EML R package overview">
<meta property="og:description" content="LivingNorwayR">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">LivingNorwayR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/EML_R_packages_overview.html">EML R package overview</a>
    </li>
    <li>
      <a href="../articles/LNWorkshopExample_TOV-E.html">Handling Darwin Core Files With Living Norway:  Example Using the TOV-E Dataset</a>
    </li>
    <li>
      <a href="../articles/LNWorkshopExample_TOV-E_Metadata.html">TOV-E Bird monitoring sampling data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LivingNorway/LivingNorwayR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="EML_R_packages_overview_files/accessible-code-block-0.0.1/empty-anchor.js"></script><script src="EML_R_packages_overview_files/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="EML_R_packages_overview_files/vis-9.1.0/vis-network.min.css" rel="stylesheet">
<script src="EML_R_packages_overview_files/vis-9.1.0/vis-network.min.js"></script><script src="EML_R_packages_overview_files/visNetwork-binding-2.1.0/visNetwork.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>EML R package overview</h1>
                        <h4 class="author">Matt</h4>
            
            <h4 class="date">15 11, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/LivingNorway/LivingNorwayR/blob/master/vignettes/EML_R_packages_overview.Rmd"><code>vignettes/EML_R_packages_overview.Rmd</code></a></small>
      <div class="hidden name"><code>EML_R_packages_overview.Rmd</code></div>

    </div>

    
    
<div id="r-packages-that-interact-with-eml" class="section level1">
<h1 class="hasAnchor">
<a href="#r-packages-that-interact-with-eml" class="anchor"></a>R packages that interact with EML</h1>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>EML (Ecological Metadata Langauage; <span class="citation">(Jones et al. 2019)</span>) is a version of the machine and human readable XML (Extensible Markup Language) which is designed to facilitate data documentation (i.e. metadata) to ease open data and sharing. XML is a language related to HTML, but was designed to tag document content and allows for validation against specific schemas. XML is well supported in many different programming languages and can be readily translated to other formats <span class="citation">(McCartney and Jones 2002)</span>.</p>
<p>EML consists of “modules” (with separate schemas) that allow the description of data attributes (e.g. spatial, temporal, taxonomic extent). EML defines several resource types (e.g. “dataset”) that inherit a common set of elements or “tags” which facilitate discovery of metadata. One of the biggest advantages of EML is that it is extensible with additional modules easily integrated <span class="citation">(McCartney and Jones 2002)</span>.</p>
<p>GBIF (the Global Biodiversity Information Facility), which is an international network and research infrastructure, provides open access to data about biodiversity. The GBIF metadata profile is based on EML and as such anyone interacting with GBIF needs to be able to read, write or visualise data in the EML format.</p>
<p>A serious barrier to data sharing is the learning curve and time cost needed to produce metadata. EML is a complex set of elements (many 100’s of them are available) most of which are not relevant to any single dataset <span class="citation">(McCartney and Jones 2002)</span>. Writing directly in EML can be a technical barrier to many researchers who, in ecology, may only have experience in the R language which has a very different syntax and data structure. The challenge is, therefore, to make filling in metadata as easy as possible for all researchers regardless of their data type.</p>
<p>Several R packages have functions to read, write or interact with EML files and here we aim to identify those that will be useful for researchers working with ecological data.</p>
<div id="aim" class="section level3">
<h3 class="hasAnchor">
<a href="#aim" class="anchor"></a>Aim</h3>
<p>To identify R packages and functions that can read, write or interact/visualise data in the EML format</p>
<div id="objectives" class="section level4">
<h4 class="hasAnchor">
<a href="#objectives" class="anchor"></a>Objectives</h4>
<ol style="list-style-type: decimal">
<li>To review the available packages on CRAN and on GitHub that have functions to interact with EML</li>
<li>To identify the main functions from these packages that allow reading, writing and visualising data in EML format</li>
</ol>
</div>
</div>
</div>
<div id="methods" class="section level2">
<h2 class="hasAnchor">
<a href="#methods" class="anchor"></a>Methods</h2>
<p>We searched for R packages on CRAN (The Comprehensive R Archive Network - <a href="https://cran.r-project.org/" class="uri">https://cran.r-project.org/</a>) using the search function which is provided by Google. We used the search term “Ecological Metadata Language site:r-project.org”. We repeated the search for “Ecological Metadata Language” on GitHub and identified repositories which used the R language. We identified all packages returned through our search that were on CRAN or on GitHub and that consisted of functions that are aimed at interacting (either reading, writing or viewing) with the Ecological Metadata Language (EML).</p>
</div>
<div id="results" class="section level2">
<h2 class="hasAnchor">
<a href="#results" class="anchor"></a>Results</h2>
<p>The search on Google (via CRAN) returned 25 hits. Seven of these were R packages. The search on GitHub returned 15 repositories of which 6 were in the R language. Two of these returned errors when we attempted to download them in RStudio using the devtools::install_github() function, and two have been archived. We identified 9 R packages (6 from CRAN and 3 from GitHub) that were active in the last two years and explicitly stated that functions were used for interacting (either reading, writing or viewing) Ecological Metadata Language (EML).</p>
<table class="table">
<colgroup>
<col width="1%">
<col width="98%">
</colgroup>
<thead><tr class="header">
<th align="left">Package</th>
<th align="left">Package_description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">EML</td>
<td align="left">Work with Ecological Metadata Language (‘EML’) files. ‘EML’ is a widely used metadata standard in the ecological and environmental sciences, described in Jones et al. (2006), <a href="doi:10.1146/annurev.ecolsys.37.091305.110031" class="uri">doi:10.1146/annurev.ecolsys.37.091305.110031</a>.</td>
</tr>
<tr class="even">
<td align="left">emld</td>
<td align="left">This is a utility for transforming Ecological Metadata Language (‘EML’) files into ‘JSON-LD’ and back into ‘EML.’ Doing so creates a list-based representation of ‘EML’ in R, so that ‘EML’ data can easily be manipulated using standard ‘R’ tools. This makes this package an effective backend for other ‘R’-based tools working with ‘EML.’ By abstracting away the complexity of ‘XML’ Schema, developers can build around native ‘R’ list objects and not have to worry about satisfying many of the additional constraints of set by the schema (such as element ordering, which is handled automatically). Additionally, the ‘JSON-LD’ representation enables the use of developer-friendly ‘JSON’ parsing and serialization that may facilitate the use of ‘EML’ in contexts outside of ‘R,’ as well as the informatics-friendly serializations such as ‘RDF’ and ‘SPARQL’ queries.</td>
</tr>
<tr class="odd">
<td align="left">RNeXML</td>
<td align="left">Provides access to phyloinformatic data in ‘NeXML’ format. The package should add new functionality to R such as the possibility to manipulate ‘NeXML’ objects in more various and refined way and compatibility with ‘ape’ objects.</td>
</tr>
<tr class="even">
<td align="left">datapack</td>
<td align="left">Provides a flexible container to transport and manipulate complex sets of data. These data may consist of multiple data files and associated meta data and ancillary files. Individual data objects have associated system level meta data, and data files are linked together using the OAI-ORE standard resource map which describes the relationships between the files. The OAI- ORE standard is described at <a href="https://www.openarchives.org/ore/" class="uri">https://www.openarchives.org/ore/</a>. Data packages can be serialized and transported as structured files that have been created following the BagIt specification. The BagIt specification is described at <a href="https://tools.ietf.org/html/draft-kunze-bagit-08" class="uri">https://tools.ietf.org/html/draft-kunze-bagit-08</a>.</td>
</tr>
<tr class="odd">
<td align="left">rbefdata</td>
<td align="left">Basic R package to access data structures offered by any BEFdata portal instance.</td>
</tr>
<tr class="even">
<td align="left">geometa</td>
<td align="left">Provides facilities to handle reading and writing of geographic metadata defined with OGC/ISO 19115, 11119 and 19110 geographic information metadata standards, and encoded using the ISO 19139 (XML) standard. It includes also a facility to check the validity of ISO 19139 XML encoded metadata.</td>
</tr>
<tr class="odd">
<td align="left">MetaEgress</td>
<td align="left">Functions to create Ecological Metadata Language (EML) XML documents from metadata stored in a metadata database design by the Long Term Ecological Research Network.</td>
</tr>
<tr class="even">
<td align="left">EMLassemblyline</td>
<td align="left">For scientists and data managers to create high quality EML metadata for dataset publication. EMLassemblyline is optimized for automating recurring publications (timeseries or data derived from timeseries sources) but works well for “one-off” publications, especially through the MetaShARK interface. EMLassemblyline prioritizes automated metadata extraction from data objects to minimize required human effort and encourages EML best practices to make publications Findable, Accessible, Interoperable, and Reusable.</td>
</tr>
<tr class="odd">
<td align="left">MetaInbase</td>
<td align="left">Use MetaInbase to convert an existing corpus of Ecological Metadata Language (EML) documents to table form. This is developed with the intention to help established LTER sites import their EML corpus into LTER-core-metabase. See <a href="https://github.com/lter/LTER-core-metabase" class="uri">https://github.com/lter/LTER-core-metabase</a>. However, other uses are possible.</td>
</tr>
<tr class="even">
<td align="left">LivingNorwayR</td>
<td align="left">The package provides a workflow for creating a Darwin Core standard-compliant data archive (“a data package”). This facilitates FAIR (Findable, Accessible, Interoperable, Reusable; <a href="https://www.go-fair.org/fair-principles/" class="uri">https://www.go-fair.org/fair-principles/</a>) data sharing and uploading of Darwin Core archives (data packages) to repositories such as GBIF. The Living Norway package also provides tools for the processing and manipulation of metadata associated with Darwin Core archives and for the import and export of metadata according to the EML (Ecological Metadata Language; <a href="https://eml.ecoinformatics.org/" class="uri">https://eml.ecoinformatics.org/</a>) standard.</td>
</tr>
</tbody>
</table>
<div id="emld-package" class="section level4">
<h4 class="hasAnchor">
<a href="#emld-package" class="anchor"></a>emld package</h4>
<p>The <em>emld</em> package <span class="citation">(Boettiger 2019)</span> is closely related to the EML package <span class="citation">(Boettiger and Jones 2021)</span> and its functions are heavily relied upon in the latest version of <em>EML</em>.</p>
<p><em>emld</em> has an advantage over <em>EML</em> where there are large, highly nested EML files. It can flatten EML in to common R formats that can be manipulated in R.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Example from https://github.com/ropensci/emld</span>
<span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/example.xml"</span>, package<span class="op">=</span><span class="st">"emld"</span><span class="op">)</span>
<span class="va">eml</span> <span class="op">&lt;-</span> <span class="fu">emld</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/emld/reference/as_emld.html">as_emld</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span>
<span class="va">eml</span><span class="op">$</span><span class="va">dataset</span><span class="op">$</span><span class="va">title</span></code></pre></div>
<pre><code>## [1] "Data from Cedar Creek LTER on productivity and species richness\n  for use in a workshop titled \"An Analysis of the Relationship between\n  Productivity and Diversity using Experimental Results from the Long-Term\n  Ecological Research Network\" held at NCEAS in September 1996."</code></pre>
<p><em>emld</em> objects are nested lists so to write EML in <em>emld</em> you can create a list.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Example from https://github.com/ropensci/emld</span>

<span class="va">me</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>individualName <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>givenName <span class="op">=</span> <span class="st">"Joe"</span>, surName <span class="op">=</span> <span class="st">"Bloggs"</span><span class="op">)</span><span class="op">)</span>

<span class="va">eml</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>dataset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
              title <span class="op">=</span> <span class="st">"The biggest fish in the sea"</span>,
              contact <span class="op">=</span> <span class="va">me</span>,
              creator <span class="op">=</span> <span class="va">me</span><span class="op">)</span>,
              system <span class="op">=</span> <span class="st">"doi"</span>,
              packageId <span class="op">=</span> <span class="st">"10.xxx"</span><span class="op">)</span>

<span class="va">ex.xml</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span><span class="op">(</span><span class="st">"ex"</span>, fileext <span class="op">=</span> <span class="st">".xml"</span><span class="op">)</span> <span class="co"># use your preferred file path</span>

<span class="fu">as_xml</span><span class="op">(</span><span class="va">eml</span>, <span class="va">ex.xml</span><span class="op">)</span></code></pre></div>
<pre><code>## NULL</code></pre>
<p><em>emld::as_xml()</em> reorders the list if necessary to ensure that it matches the EML required format. You can use <em>emld::eml_validate()</em> to check that the EML has been written correctly and is a valid EML file.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">eml_validate</span><span class="op">(</span><span class="va">ex.xml</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] TRUE
## attr(,"errors")
## character(0)</code></pre>
</div>
<div id="eml-package" class="section level4">
<h4 class="hasAnchor">
<a href="#eml-package" class="anchor"></a>EML package</h4>
<p>The <em>EML</em> <span class="citation">(Boettiger and Jones 2021)</span> package uses <em>emld</em> <span class="citation">(Boettiger 2019)</span> as a basis for many of its functions.</p>
<p>The read_eml() function reads eml files in to the R session.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"example.xml"</span>, package <span class="op">=</span> <span class="st">"emld"</span><span class="op">)</span>
<span class="va">eml</span> <span class="op">&lt;-</span> <span class="fu">EML</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/EML/reference/read_eml.html">read_eml</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span>
<span class="va">eml</span><span class="op">$</span><span class="va">dataset</span><span class="op">$</span><span class="va">title</span></code></pre></div>
<pre><code>## [1] "Data from Cedar Creek LTER on productivity and species richness\n  for use in a workshop titled \"An Analysis of the Relationship between\n  Productivity and Diversity using Experimental Results from the Long-Term\n  Ecological Research Network\" held at NCEAS in September 1996."</code></pre>
<p>Several functions (with the prefix <em>set_</em>) allow the user to import attributes of the dataset file (e.g. the methods) as word or Markdown documents and merge them in to a single EML file (see <a href="https://docs.ropensci.org/EML/" class="uri">https://docs.ropensci.org/EML/</a> for a detailed tutorial).</p>
<p><em>EML</em> has several functions (with the prefix <em>get_</em>) which allow a user to extract elements of an XML file.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"tests"</span>, <span class="fu">emld</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/emld/reference/eml_version.html">eml_version</a></span><span class="op">(</span><span class="op">)</span>, 
  <span class="st">"eml-datasetWithAttributelevelMethods.xml"</span>, package <span class="op">=</span> <span class="st">"emld"</span><span class="op">)</span>
<span class="va">eml</span> <span class="op">&lt;-</span> <span class="fu">EML</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/EML/reference/read_eml.html">read_eml</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span>
<span class="fu">EML</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/EML/reference/get_attributes.html">get_attributes</a></span><span class="op">(</span><span class="va">eml</span><span class="op">$</span><span class="va">dataset</span><span class="op">$</span><span class="va">dataTable</span><span class="op">$</span><span class="va">attributeList</span><span class="op">)</span></code></pre></div>
<pre><code>## $attributes
##    id...1 attributeName                    attributeLabel
## 1   att.1           fld                             Field
## 2   att.2          year                              year
## 3   att.3            sr                  Species Richness
## 4   att.4        pctcov                     percent cover
## 5   att.5       avesr91 Average Species Richness for 1991
## 6    &lt;NA&gt;       avesr92 Average Species Richness for 1992
## 7    &lt;NA&gt;       avesr93 Average Species Richness for 1993
## 8    &lt;NA&gt;       avesr94 Average Species Richness for 1994
## 9    &lt;NA&gt;       avesr95 Average Species Richness for 1995
## 10   &lt;NA&gt;       avesr96 Average Species Richness for 1996
## 11   &lt;NA&gt;        MeanSR             mean species richness
## 12 att.14          time                              Time
##                                              attributeDefinition storageType
## 1                   Field where the data was collected\n              string
## 2                      The year the data was collected\n               gYear
## 3                                       Species richness for CDR       float
## 4                The percent ground cover on the field\n               float
## 5   The average species richness for the field in 1991\n               float
## 6   The average species richness for the field in 1992\n               float
## 7   The average species richness for the field in 1993\n               float
## 8   The average species richness for the field in 1994\n               float
## 9   The average species richness for the field in 1995\n               float
## 10  The average species richness for the field in 1996\n               float
## 11         the mean species richness from 1991 to 1996\n               float
## 12 The time of day for this observation, 24 hour clock\n                time
##    id...6            definition measurementScale         domain formatString
## 1    nd.1 Valid names of fields          nominal     textDomain         &lt;NA&gt;
## 2    &lt;NA&gt;                  &lt;NA&gt;         dateTime dateTimeDomain         YYYY
## 3    &lt;NA&gt;                  &lt;NA&gt;         interval  numericDomain         &lt;NA&gt;
## 4    &lt;NA&gt;                  &lt;NA&gt;            ratio  numericDomain         &lt;NA&gt;
## 5    &lt;NA&gt;                  &lt;NA&gt;            ratio  numericDomain         &lt;NA&gt;
## 6    &lt;NA&gt;                  &lt;NA&gt;            ratio  numericDomain         &lt;NA&gt;
## 7    &lt;NA&gt;                  &lt;NA&gt;            ratio  numericDomain         &lt;NA&gt;
## 8    &lt;NA&gt;                  &lt;NA&gt;            ratio  numericDomain         &lt;NA&gt;
## 9    &lt;NA&gt;                  &lt;NA&gt;            ratio  numericDomain         &lt;NA&gt;
## 10   &lt;NA&gt;                  &lt;NA&gt;            ratio  numericDomain         &lt;NA&gt;
## 11   &lt;NA&gt;                  &lt;NA&gt;            ratio  numericDomain         &lt;NA&gt;
## 12   &lt;NA&gt;                  &lt;NA&gt;         dateTime dateTimeDomain   hh:mm:ss.s
##    dateTimePrecision id...8    minimum          unit precision numberType
## 1               &lt;NA&gt;   &lt;NA&gt;       &lt;NA&gt;          &lt;NA&gt;      &lt;NA&gt;       &lt;NA&gt;
## 2                  1   dd.2       1944          &lt;NA&gt;      &lt;NA&gt;       &lt;NA&gt;
## 3               &lt;NA&gt;   nd.3          0 dimensionless       0.5       real
## 4               &lt;NA&gt;   nd.4          0 dimensionless       0.1       real
## 5               &lt;NA&gt;   nd.5          0 dimensionless       0.1       real
## 6               &lt;NA&gt;   &lt;NA&gt;       &lt;NA&gt; dimensionless       0.1       &lt;NA&gt;
## 7               &lt;NA&gt;   &lt;NA&gt;       &lt;NA&gt; dimensionless       0.1       &lt;NA&gt;
## 8               &lt;NA&gt;   &lt;NA&gt;       &lt;NA&gt; dimensionless       0.1       &lt;NA&gt;
## 9               &lt;NA&gt;   &lt;NA&gt;       &lt;NA&gt; dimensionless       0.1       &lt;NA&gt;
## 10              &lt;NA&gt;   &lt;NA&gt;       &lt;NA&gt; dimensionless       0.1       &lt;NA&gt;
## 11              &lt;NA&gt;   &lt;NA&gt;       &lt;NA&gt; dimensionless       0.1       &lt;NA&gt;
## 12               0.1   dd.3 15:00:00.0          &lt;NA&gt;      &lt;NA&gt;       &lt;NA&gt;
##    exclusive...10 exclusive...12    maximum exclusive...9 exclusive...11
## 1            &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;          &lt;NA&gt;           &lt;NA&gt;
## 2            &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;          &lt;NA&gt;           &lt;NA&gt;
## 3            &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;          &lt;NA&gt;           &lt;NA&gt;
## 4            true           true        100          &lt;NA&gt;           &lt;NA&gt;
## 5            &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;          &lt;NA&gt;           &lt;NA&gt;
## 6            &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;          &lt;NA&gt;           &lt;NA&gt;
## 7            &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;          &lt;NA&gt;           &lt;NA&gt;
## 8            &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;          &lt;NA&gt;           &lt;NA&gt;
## 9            &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;          &lt;NA&gt;           &lt;NA&gt;
## 10           &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;          &lt;NA&gt;           &lt;NA&gt;
## 11           &lt;NA&gt;           &lt;NA&gt;       &lt;NA&gt;          &lt;NA&gt;           &lt;NA&gt;
## 12           &lt;NA&gt;           &lt;NA&gt; 19:00:00.0          true           true
## 
## $factors
## NULL</code></pre>
</div>
<div id="rnexml-package" class="section level3">
<h3 class="hasAnchor">
<a href="#rnexml-package" class="anchor"></a>RNeXML package</h3>
<p><em>RNeXML</em> <span class="citation">(Boettiger et al. 2016)</span> is a package for reading and writing phylogenetic, character and trait metadata focussed on taxonomy. The package stands parallel to <em>EML</em> and <em>emd</em> using similar functions but is focused on a different (but related) standard NeXML rather than EML per se.</p>
</div>
<div id="datapack-package" class="section level3">
<h3 class="hasAnchor">
<a href="#datapack-package" class="anchor"></a>datapack package</h3>
<p>The <em>datapack</em> package <span class="citation">(Jones and Slaughter 2020)</span> provides functions for collating multiple data and metadata objects of different types into a bundle that can be transported and loaded using a single composite file. It is primarily meant as a container to bundle together files for transport to or from DataONE data repositories. Metadata in the form of EML can be attached to the databundle.</p>
<pre><code>## Members:
## 
## filename       format    mediaType  size     identifier          modified local 
## sample-eml.xml eml...1.0 NA         5990     urn:uuid...a890a5ef n        y     
## 
## Package identifier: NA
## RightsHolder: NA
## 
## 
## This package does not contain any provenance relationships.</code></pre>
</div>
<div id="rbefdata-package" class="section level3">
<h3 class="hasAnchor">
<a href="#rbefdata-package" class="anchor"></a>rbefdata package</h3>
<p>Similarly to the <em>datapack</em> package <em>rbefdata</em> <span class="citation">(Pfaff, Nadrowski, and Man 2013)</span> links to the BEF data portal (<a href="https://fundiv.befdata.biow.uni-leipzig.de/" class="uri">https://fundiv.befdata.biow.uni-leipzig.de/</a>). The function <em>rbefdata::bef.portal.get.metadata()</em> extracts the metadata from a file that the user has downloaded from the BEF data portal.</p>
</div>
<div id="geometa-package" class="section level3">
<h3 class="hasAnchor">
<a href="#geometa-package" class="anchor"></a>geometa package</h3>
<p><em>geometa</em> <span class="citation">(Blondel 2021)</span> provides functions for reading and writing geographic metadata. It suggests <em>EML</em> and <em>emld</em>. It can convert metadata to and from EML using the <em>geometa::convert_metadata()</em> function.</p>
</div>
<div id="metaegress-package" class="section level3">
<h3 class="hasAnchor">
<a href="#metaegress-package" class="anchor"></a>MetaEgress package</h3>
<p><em>MetaEgress</em> <span class="citation">(Nguyen and Kui 2021)</span> is designed to create EML for Long Term Ecological Research metabase ( <a href="https://github.com/lter/LTER-core-metabase" class="uri">https://github.com/lter/LTER-core-metabase</a>). Validation of EML is done through <em>emld::eml_validate()</em></p>
</div>
<div id="emlassemblyline-package" class="section level3">
<h3 class="hasAnchor">
<a href="#emlassemblyline-package" class="anchor"></a>EMLassemblyline package</h3>
<p><em>EMLassemblyline</em> <span class="citation">(Smith 2021)</span> is a metadata builder that has functions to autoextract metadata. The package is centered around a “data package” that is a collection of data objects and metadata. The templating functions allow the development of metadata templates that the user can add information to. <em>EMLassemblyline</em> also has a “living data” set of functions that allow ongoing data collection to be published at regular periods. See <a href="https://ediorg.github.io/EMLassemblyline/index.html" class="uri">https://ediorg.github.io/EMLassemblyline/index.html</a> for examples of the workflow.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/EDIorg/EMLassemblyline">EMLassemblyline</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/EMLassemblyline/man/make_eml.html">make_eml</a></span><span class="op">(</span>
  path <span class="op">=</span> <span class="st">"./metadata_templates"</span>,
  data.path <span class="op">=</span> <span class="st">"./data_objects"</span>,
  eml.path <span class="op">=</span> <span class="st">"./eml"</span>,
  dataset.title <span class="op">=</span> <span class="st">"Sphagnum and Vascular Plant Decomposition under Increasing Nitrogen Additions"</span>,
  temporal.coverage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"2014-05-01"</span>, <span class="st">"2015-10-31"</span><span class="op">)</span>,
  maintenance.description <span class="op">=</span> <span class="st">"Completed: No updates to these data are expected"</span>,
  data.table <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"decomp.csv"</span>, <span class="st">"nitrogen.csv"</span><span class="op">)</span>,
  data.table.description <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Decomposition data"</span>, <span class="st">"Nitrogen data"</span><span class="op">)</span>,
  other.entity <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ancillary_data.zip"</span>, <span class="st">"processing_and_analysis.R"</span><span class="op">)</span>,
  other.entity.description <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Ancillary data"</span>, <span class="st">"Data processing and analysis script"</span><span class="op">)</span>,
  user.id <span class="op">=</span> <span class="st">"myid"</span>,
  user.domain <span class="op">=</span> <span class="st">"EDI"</span>,
  package.id <span class="op">=</span> <span class="st">"edi.260.1"</span><span class="op">)</span></code></pre></div>
</div>
<div id="metainbase-package" class="section level3">
<h3 class="hasAnchor">
<a href="#metainbase-package" class="anchor"></a>MetaInbase package</h3>
<p><em>MetaInbase</em> <span class="citation">(c)) 2021)</span> has functions to convert EML to tables. It appears to be in an early stage of development</p>
</div>
<div id="metashark-package" class="section level3">
<h3 class="hasAnchor">
<a href="#metashark-package" class="anchor"></a>MetaShARK package</h3>
<p><em>MetaShARK</em> (Metadata Shiny Automated Resource &amp; Knowledge; <a href="https://github.com/earnaud/MetaShARK-v2" class="uri">https://github.com/earnaud/MetaShARK-v2</a>) is a R shiny app allowing the user to get information about EML and to fill in metadata for datasets according to this standard. This package has developed a “user-friendly” Shiny App for researchers not familiar with EML standards.</p>
</div>
<div id="livingnorwayr" class="section level3">
<h3 class="hasAnchor">
<a href="#livingnorwayr" class="anchor"></a>LivingNorwayR</h3>
<p><em>LivingNorwayR</em> addresses the EML metadata in a slightly different way to the other packages. It uses RMarkdown with specific tagging functions in order to “knit” together the EML document. The package is focused on creating Darwin Core compliant data archives (“data packages”) that can be stored locally or shared with, for example, GBIF. A worked example is available as a <a href="https://raw.githubusercontent.com/LivingNorway/LivingNorwayR/master/vignettes/LNWorkshopExample_TOV-E_Metadata.rmd">vignette</a>.</p>
</div>
</div>
<div id="relationship-between-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#relationship-between-packages" class="anchor"></a>Relationship between packages</h2>
<p>By looking at the dependency relationship between the packages we can get an understanding of how they are structured and how similar each package is. From the figure and table below you can see that <em>jsonlite</em> and XML2 are core packages that underpin many of the packages that deal with EML in R. <em>jsonlite</em> helps to convert the list-like structure of EML in to different R classes. XML2 allows the user to read in XML files (EML is a type of XML) to R as a list and to also write EML as an XML structured file. Three of the packages use the <em>emld</em> package as a basis (most use the validation function) and from the network plot you can see that all the packages except <em>EMLassemblyline</em> are quite closely related in terms of their underlying dependencies. A large number of the dependencies of <em>EMLassemblyline</em> are not shared by the other packages.</p>
<div id="htmlwidget-f53a4be88dae141d87fa" style="width:100%;height:432.632880098888px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-f53a4be88dae141d87fa">{"x":{"nodes":{"id":["EML","emld","RNeXML","datapack","rbefdata","geometa","MetaEgress","EMLassemblyline","R6","Rcpp","V8","base64enc","cli","crayon","curl","digest","dplyr","ellipsis","evaluate","fansi","fastmap","generics","glue","grDevices","graphics","highr","htmltools","jqr","jquerylib","jsonld","jsonlite","knitr","lazyeval","lifecycle","magrittr","methods","pillar","pkgconfig","purrr","rlang","rmarkdown","stats","stringi","stringr","tibble","tidyselect","tinytex","tools","utf8","utils","uuid","vctrs","xfun","xml2","yaml","XML","ape","askpass","compiler","grid","httr","lattice","mime","nlme","openssl","parallel","plyr","reshape2","sys","tidyr","brew","commonmark","desc","hash","pkgload","redland","roxygen2","rprojroot","rstudioapi","withr","RColorBrewer","RCurl","bitops","gdata","gtools","rjson","rtematres","wordcloud","DBI","RPostgres","bit","bit64","blob","clipr","hms","lubridate","readr","tzdb","vroom","EDIutils","Formula","Hmisc","MASS","Matrix","NCmisc","WikidataQueryServiceR","WikidataR","WikipediR","assertthat","backports","bold","checkmate","cluster","codetools","colorspace","conditionz","covr","crul","data.table","farver","fastmatch","foreach","foreign","ggplot2","gridExtra","gtable","htmlTable","htmlwidgets","httpcode","igraph","isoband","iterators","jpeg","labeling","latticeExtra","mgcv","munsell","natserv","nnet","pbapply","phangorn","png","prettyunits","proftools","progress","quadprog","ratelimitr","reader","rentrez","reshape","rex","ritis","rncl","rotl","rpart","rredlist","scales","solrium","splines","survival","taxize","taxonomyCleanr","triebeard","urltools","viridis","viridisLite","wikitaxa","worrms","zoo"],"label":["EML","emld","RNeXML","datapack","rbefdata","geometa","MetaEgress","EMLassemblyline","R6","Rcpp","V8","base64enc","cli","crayon","curl","digest","dplyr","ellipsis","evaluate","fansi","fastmap","generics","glue","grDevices","graphics","highr","htmltools","jqr","jquerylib","jsonld","jsonlite","knitr","lazyeval","lifecycle","magrittr","methods","pillar","pkgconfig","purrr","rlang","rmarkdown","stats","stringi","stringr","tibble","tidyselect","tinytex","tools","utf8","utils","uuid","vctrs","xfun","xml2","yaml","XML","ape","askpass","compiler","grid","httr","lattice","mime","nlme","openssl","parallel","plyr","reshape2","sys","tidyr","brew","commonmark","desc","hash","pkgload","redland","roxygen2","rprojroot","rstudioapi","withr","RColorBrewer","RCurl","bitops","gdata","gtools","rjson","rtematres","wordcloud","DBI","RPostgres","bit","bit64","blob","clipr","hms","lubridate","readr","tzdb","vroom","EDIutils","Formula","Hmisc","MASS","Matrix","NCmisc","WikidataQueryServiceR","WikidataR","WikipediR","assertthat","backports","bold","checkmate","cluster","codetools","colorspace","conditionz","covr","crul","data.table","farver","fastmatch","foreach","foreign","ggplot2","gridExtra","gtable","htmlTable","htmlwidgets","httpcode","igraph","isoband","iterators","jpeg","labeling","latticeExtra","mgcv","munsell","natserv","nnet","pbapply","phangorn","png","prettyunits","proftools","progress","quadprog","ratelimitr","reader","rentrez","reshape","rex","ritis","rncl","rotl","rpart","rredlist","scales","solrium","splines","survival","taxize","taxonomyCleanr","triebeard","urltools","viridis","viridisLite","wikitaxa","worrms","zoo"]},"edges":{"from":["R6","Rcpp","V8","base64enc","cli","crayon","curl","digest","dplyr","ellipsis","emld","evaluate","fansi","fastmap","generics","glue","grDevices","graphics","highr","htmltools","jqr","jquerylib","jsonld","jsonlite","knitr","lazyeval","lifecycle","magrittr","methods","pillar","pkgconfig","purrr","rlang","rmarkdown","stats","stringi","stringr","tibble","tidyselect","tinytex","tools","utf8","utils","uuid","vctrs","xfun","xml2","yaml","Rcpp","V8","curl","grDevices","graphics","jsonld","jsonlite","methods","stats","utils","xml2","yaml","R6","Rcpp","XML","ape","askpass","cli","compiler","crayon","curl","dplyr","ellipsis","fansi","generics","glue","grDevices","graphics","grid","httr","jsonlite","lattice","lazyeval","lifecycle","magrittr","methods","mime","nlme","openssl","parallel","pillar","pkgconfig","plyr","purrr","reshape2","rlang","stats","stringi","stringr","sys","tibble","tidyr","tidyselect","tools","utf8","utils","uuid","vctrs","xml2","R6","XML","brew","cli","commonmark","crayon","desc","digest","evaluate","glue","grDevices","graphics","hash","highr","knitr","magrittr","methods","pkgload","purrr","redland","rlang","roxygen2","rprojroot","rstudioapi","stats","stringi","stringr","tools","utils","uuid","withr","xfun","xml2","yaml","RColorBrewer","RCurl","Rcpp","XML","bitops","gdata","grDevices","graphics","gtools","methods","plyr","rjson","rtematres","stats","utils","wordcloud","R6","XML","askpass","curl","grDevices","graphics","httr","jsonlite","methods","mime","openssl","stats","sys","tools","utils","DBI","EML","R6","RPostgres","Rcpp","V8","base64enc","bit","bit64","blob","cli","clipr","crayon","curl","digest","dplyr","ellipsis","emld","evaluate","fansi","fastmap","generics","glue","grDevices","graphics","highr","hms","htmltools","jqr","jquerylib","jsonld","jsonlite","knitr","lazyeval","lifecycle","lubridate","magrittr","methods","pillar","pkgconfig","purrr","readr","rlang","rmarkdown","stats","stringi","stringr","tibble","tidyselect","tinytex","tools","tzdb","utf8","utils","uuid","vctrs","vroom","withr","xfun","xml2","yaml","EDIutils","EML","Formula","Hmisc","MASS","Matrix","NCmisc","R6","RColorBrewer","Rcpp","V8","WikidataQueryServiceR","WikidataR","WikipediR","XML","ape","askpass","assertthat","backports","base64enc","bit","bit64","bold","checkmate","cli","clipr","cluster","codetools","colorspace","compiler","conditionz","covr","crayon","crul","curl","data.table","digest","dplyr","ellipsis","emld","evaluate","fansi","farver","fastmap","fastmatch","foreach","foreign","generics","ggplot2","glue","grDevices","graphics","grid","gridExtra","gtable","highr","hms","htmlTable","htmltools","htmlwidgets","httpcode","httr","igraph","isoband","iterators","jpeg","jqr","jquerylib","jsonld","jsonlite","knitr","labeling","lattice","latticeExtra","lazyeval","lifecycle","lubridate","magrittr","methods","mgcv","mime","munsell","natserv","nlme","nnet","openssl","parallel","pbapply","phangorn","pillar","pkgconfig","plyr","png","prettyunits","proftools","progress","purrr","quadprog","ratelimitr","reader","readr","rentrez","reshape","rex","ritis","rlang","rmarkdown","rncl","rotl","rpart","rredlist","rstudioapi","scales","solrium","splines","stats","stringi","stringr","survival","sys","taxize","taxonomyCleanr","tibble","tidyr","tidyselect","tinytex","tools","triebeard","tzdb","urltools","utf8","utils","uuid","vctrs","viridis","viridisLite","vroom","wikitaxa","withr","worrms","xfun","xml2","yaml","zoo"],"to":["EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","EML","emld","emld","emld","emld","emld","emld","emld","emld","emld","emld","emld","emld","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","RNeXML","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","datapack","rbefdata","rbefdata","rbefdata","rbefdata","rbefdata","rbefdata","rbefdata","rbefdata","rbefdata","rbefdata","rbefdata","rbefdata","rbefdata","rbefdata","rbefdata","rbefdata","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","geometa","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","MetaEgress","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline","EMLassemblyline"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot"},"manipulation":{"enabled":false},"interaction":{"dragNodes":false,"dragView":false,"hover":true,"navigationButtons":true,"zoomSpeed":1},"edges":{"arrows":{"to":{"enabled":true,"scaleFactor":2}},"color":{"color":"lightblue","highlight":"red"}},"physics":{"solver":"forceAtlas2Based","forceAtlas2Based":{"gravitationalConstant":-300}}},"groups":null,"width":"100%","height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","tooltipStay":300,"tooltipStyle":"position: fixed;visibility:hidden;padding: 5px;white-space: nowrap;font-family: verdana;font-size:14px;font-color:#000000;background-color: #f5f4ed;-moz-border-radius: 3px;-webkit-border-radius: 3px;border-radius: 3px;border: 1px solid #808074;box-shadow: 3px 3px 10px rgba(0, 0, 0, 0.2);"},"evals":[],"jsHooks":[]}</script><div id="which-package-to-use" class="section level3">
<h3 class="hasAnchor">
<a href="#which-package-to-use" class="anchor"></a>Which package to use?</h3>
<p>Many of the packages we have found are focused on single data repositories. These packages might be useful for these specific tasks, however the main packages for creating EML are {emld} and {EML} as these are the basis to many of the functions used in the other packages. The {LivingNorwayR} package uses RMarkdown as a tool to create EML metadata and might be preferable to those with some experience with RMarkdown documents.</p>
</div>
</div>
<div id="places-to-go-for-help" class="section level2">
<h2 class="hasAnchor">
<a href="#places-to-go-for-help" class="anchor"></a>Places to go for help</h2>
<ul>
<li><a href="https://environmentaldatainitiative.org/five-phases-of-data-publishing/phase-3/make-eml-with-r/" class="uri">https://environmentaldatainitiative.org/five-phases-of-data-publishing/phase-3/make-eml-with-r/</a></li>
<li><a href="https://cran.r-project.org/web/packages/EML/vignettes/creating-EML.html" class="uri">https://cran.r-project.org/web/packages/EML/vignettes/creating-EML.html</a></li>
<li><a href="https://cran.r-project.org/web/packages/emld/vignettes/tutorial.html" class="uri">https://cran.r-project.org/web/packages/emld/vignettes/tutorial.html</a></li>
<li><a href="https://www.neonscience.org/dc-metadata-importance-eml-r" class="uri">https://www.neonscience.org/dc-metadata-importance-eml-r</a></li>
<li><a href="https://ropensci.org/blog/2017/08/01/emldown/" class="uri">https://ropensci.org/blog/2017/08/01/emldown/</a></li>
</ul>
<div id="contributors" class="section level4">
<h4 class="hasAnchor">
<a href="#contributors" class="anchor"></a>Contributors</h4>
<p>Matt Grainger</p>
</div>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs">
<div id="ref-geometa">
<p>Blondel, Emmanuel. 2021. <em>Geometa: Tools for Reading and Writing Iso/Ogc Geographic Metadata</em>. <a href="https://CRAN.R-project.org/package=geometa">https://CRAN.R-project.org/package=geometa</a>.</p>
</div>
<div id="ref-emld">
<p>Boettiger, Carl. 2019. “Ecological Metadata as Linked Data. Journal of Open Source Software.” <em>The Journal of Open Source Software</em> 4 (34): 1276. <a href="https://doi.org/10.21105/joss.01276">https://doi.org/10.21105/joss.01276</a>.</p>
</div>
<div id="ref-RNeXML">
<p>Boettiger, Carl, Scott Chamberlain, Rutger Vos, and Hilmar Lapp. 2016. “RNeXML: A Package for Reading and Writing Richly Annotated Phylogenetic, Character, and Trait Data in R.” <em>Methods in Ecology and Evolution</em> 7: 352–57. <a href="https://doi.org/10.1111/2041-210X.12469">https://doi.org/10.1111/2041-210X.12469</a>.</p>
</div>
<div id="ref-EML">
<p>Boettiger, Carl, and Matthew B. Jones. 2021. <em>EML: Read and Write Ecological Metadata Language Files</em>. <a href="https://CRAN.R-project.org/package=EML">https://CRAN.R-project.org/package=EML</a>.</p>
</div>
<div id="ref-MetaInbase">
<p>c)). 2021. <em>MetaInbase: Convert Corpus of Ecological Metadata Language (Eml) Documents to Table Form</em>.</p>
</div>
<div id="ref-datapack">
<p>Jones, Matthew B., and Peter Slaughter. 2020. <em>datapack: A Flexible Container to Transport and Manipulate Data and Associated Resources</em>. <a href="https://doi.org/10.5063/F1QV3JGM">https://doi.org/10.5063/F1QV3JGM</a>.</p>
</div>
<div id="ref-EML_2019">
<p>Jones, Matthew, Margaret O’Brien, Bryce Mecum, Carl Boettiger, Mark Schildhauer, Mitchell Maier, Timothy Whiteaker, Stevan Earl, and Steven Chong. 2019. “Ecological Metadata Language Version 2.2.0.” <a href="https://doi.org/10.5063/f11834t2">https://doi.org/10.5063/f11834t2</a>.</p>
</div>
<div id="ref-mccartney2002using">
<p>McCartney, P, and M Jones. 2002. “Using Xml-Encoded Metadata as a Basis for Advanced Information Systems for Ecological Research.” In <em>Proc. 6th World Multiconference Systemics, Cybernetics and Informatics</em>, 7:379–84.</p>
</div>
<div id="ref-MetaEgress">
<p>Nguyen, An T., and Li Kui. 2021. <em>MetaEgress: Create Ecological Metadata Language from Lter Core Metabase</em>.</p>
</div>
<div id="ref-rbefdata">
<p>Pfaff, Claas-Thido, Karin Nadrowski, and Xingxing Man. 2013. <em>Rbefdata: BEFdata R Package</em>. <a href="https://CRAN.R-project.org/package=rbefdata">https://CRAN.R-project.org/package=rbefdata</a>.</p>
</div>
<div id="ref-EMLassemblyline">
<p>Smith, Colin. 2021. <em>EMLassemblyline: A Tool Kit for Building Eml Metadata Workflows</em>. <a href="https://github.com/EDIorg/EMLassemblyline">https://github.com/EDIorg/EMLassemblyline</a>.</p>
</div>
</div>
<div style="page-break-after: always;"></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Joseph Chipperfield, Matthew Grainger, Erlend Nilsen, Norsk institutt for naturforskning (NINA).</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
